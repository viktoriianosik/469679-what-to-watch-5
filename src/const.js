export const MAX_STAR_COUNT = 5;
export const SIMILAR_MOVIES_COUNT = 4;
export const MOVIES_COUNT = 8;

export const TABS = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};
