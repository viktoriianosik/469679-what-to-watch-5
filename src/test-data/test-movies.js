export const testMovies = [
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 1,
    isFavorite: false,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 1.5,
    released: 1994,
    runTime: 153,
    scoresCount: 1635992,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  },
  {
    backgroundColor: `#BDAD8F`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/No_Country_for_Old_Men.jpg`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    director: `Ethan Coen`,
    genre: `Crime`,
    id: 14,
    isFavorite: false,
    name: `No Country for Old Men`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/No_Country_for_Old_Men.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/no-country-for-old-men.jpg`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 4.1,
    released: 2007,
    runTime: 122,
    scoresCount: 764976,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  }
];
