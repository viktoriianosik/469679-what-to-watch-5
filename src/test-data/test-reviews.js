export const testReviews = [
  {
    comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2020-10-09T13:38:44.769Z`,
    id: 1,
    rating: 3.3,
    user: {
      id: 19,
      name: `Christina`
    },
  },
  {
    comment: `I love this movie.`,
    date: `2020-13-09T08:38:44.769Z`,
    id: 1,
    rating: 5,
    user: {
      id: 9,
      name: `Maks`
    },
  },
  {
    comment: `Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2019-30-08T13:38:44.769Z`,
    id: 14,
    rating: 8.3,
    user: {
      id: 3,
      name: `Olga`
    },
  },
];
