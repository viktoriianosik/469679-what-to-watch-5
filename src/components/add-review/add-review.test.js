import React from "react";
import renderer from "react-test-renderer";
import AddReview from "./add-review";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {testMovies} from "../../test-data/test-movies";
import {MemoryRouter} from "react-router-dom";
import {AuthorizationStatus} from "../../const";

const noop = () => {};

describe(`Render AddReview correctly`, () => {
  describe(`Render AddReview correctly without error message`, () => {
    const mockStore = configureStore([]);
    let store = null;
    let addReviewComponent = null;

    beforeEach(() => {
      store = mockStore({
        DATA: {
          movie: testMovies[0],
        },
        ERROR: {
          errorMessage: ``,
        },
        USER: {
          authorizationStatus: AuthorizationStatus.AUTH,
        }
      });

      store.dispatch = jest.fn();

      addReviewComponent = renderer.create(
          <Provider store={store}>
            <MemoryRouter keyLength={0}>
              <AddReview
                userStars={[true, true, true, true, true]}
                onInputChange={noop}
                onTextChange={noop}
                onSubmit={noop}
                movieID={`1`}
              />
            </MemoryRouter>
          </Provider>
      );
    });

    it(``, () => {
      expect(addReviewComponent.toJSON()).toMatchSnapshot();
    });
  });

  describe(`Render AddReview correctly with error message`, () => {
    const mockStore = configureStore([]);
    let store = null;
    let addReviewComponent = null;

    beforeEach(() => {
      store = mockStore({
        DATA: {
          movie: testMovies[0],
        },
        ERROR: {
          errorMessage: `error 404`,
        },
        USER: {
          authorizationStatus: AuthorizationStatus.AUTH,
        }
      });

      store.dispatch = jest.fn();

      addReviewComponent = renderer.create(
          <Provider store={store}>
            <MemoryRouter keyLength={0}>
              <AddReview
                userStars={[true, true, true, true, true]}
                onInputChange={noop}
                onTextChange={noop}
                onSubmit={noop}
                movieID={`1`}
              />
            </MemoryRouter>
          </Provider>
      );
    });

    it(``, () => {
      expect(addReviewComponent.toJSON()).toMatchSnapshot();
    });
  });
});
