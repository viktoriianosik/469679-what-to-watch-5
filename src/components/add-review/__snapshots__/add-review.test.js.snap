// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AddReview correctly Render AddReview correctly with error message  1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "background": "#795433",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "background": "red",
        "color": "white",
        "display": "flex",
        "height": "70px",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
        "zIndex": "100",
      }
    }
  >
    error 404
  </div>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              Pulp Fiction
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          maxLength="400"
          minLength="50"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;

exports[`Render AddReview correctly Render AddReview correctly without error message  1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "background": "#795433",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              Pulp Fiction
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="The Grand Budapest Hotel poster"
        height="327"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 
            1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 
            2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 
            3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 
            4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            onChange={[Function]}
            type="radio"
            value={5}
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 
            5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          maxLength="400"
          minLength="50"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
